src/openMVG/features/regions.hpp

```diff
@@ -78,8 +106,15 @@ class Regions
 
   virtual Regions * EmptyClone() const = 0;
 
+  virtual std::unique_ptr<Regions> createFilteredRegions(
+                     const std::vector<FeatureInImage>& featuresInImage,
+                     std::vector<IndexT>& out_associated3dPoint,
+                     std::map<IndexT, IndexT>& out_mapFullToLocal) const = 0;
+
 };
 
+inline Regions::~Regions() {}
```

Comment:
Yes, but the destructor is a special function, you always have to provide an implementation.